tap_bing_ads/__init__.py

```diff
@@ -1037,12 +1035,61 @@ def build_report_request(
     await asyncio.gather(*sync_account_data_tasks)
 
 
+def refresh_access_token():
+    authentication = OAuthWebAuthCodeGrant(
+        CONFIG["oauth_client_id"], CONFIG["oauth_client_secret"], ""
+    )
+    return authentication.request_oauth_tokens_by_refresh_token(
+        CONFIG["refresh_token"]
+    )._access_token
+
+
+def get_accounts_customer():
+    access_token = refresh_access_token()
+    TOPN = 1
+    url = "https://clientcenter.api.bingads.microsoft.com/Api/CustomerManagement/v13/CustomerManagementService.svc"
+    headers = {"content-type": "text/xml", "SOAPAction": "FindAccountsOrCustomersInfo"}
+    body = f"""<s:Envelope xmlns:i="http://www.w3.org/2001/XMLSchema-instance" xmlns:s="http://schemas.xmlsoap.org/soap/envelope/">
+    <s:Header xmlns="https://bingads.microsoft.com/Customer/v13">
+        <Action mustUnderstand="1">FindAccountsOrCustomersInfo</Action>
+        <AuthenticationToken i:nil="false">{access_token}</AuthenticationToken>
+        <DeveloperToken i:nil="false">{CONFIG['developer_token']}</DeveloperToken>
+    </s:Header>
+    <s:Body>
+        <FindAccountsOrCustomersInfoRequest xmlns="https://bingads.microsoft.com/Customer/v13">
+            <CustomerNameFilter i:nil="false">" "</CustomerNameFilter>
+            <TopN>{TOPN}</TopN>
+        </FindAccountsOrCustomersInfoRequest>
+    </s:Body>
+    </s:Envelope>"""
+
+    try:
+        response = requests.post(url, data=body, headers=headers)
+        response.raise_for_status()
+    except HTTPError as http_err:
+        LOGGER.error(f"HTTP error occurred: {http_err}")
+    except Exception as err:
+        LOGGER.error(f"Other error occurred: {err}")
+    response = response.content.decode("utf-8")
+    customer_id = response[
+        response.find("<a:CustomerId>") + 14 : response.find("</a:CustomerId>")
+    ]
+    account_ids = response[
+        response.find("<a:AccountId>") + 13 : response.find("</a:AccountId>")
+    ]
```

Comment:
Check if there's a library that parses SOAP cause the way we are looking at now it's unsafe